README.md

```diff
@@ -47,59 +58,156 @@ To run the Flask application correctly, it requires some environment variables t
 
 ### SECRET_KEY
+
 - **Description:** The secret key used for creating encrypted tokens and for the Flask session.
 - **Recommendation:** Generate a strong random value for this.
 
 ### JWT_ACCESS_TOKEN_EXPIRES_MINUTES
+
 - **Description:** The expiration time (in minutes) for the JWT access token.
 - **Example:** `JWT_ACCESS_TOKEN_EXPIRES_MINUTES=30`
 
 ### SESSION_TYPE
-- **Description:** Specifies where to store the session data. Options can include 'filesystem', 'redis', 'memcached', etc. 
+
+- **Description:** Specifies where to store the session data. Options can include 'filesystem', 'redis', 'memcached', etc.
 - **Example:** `SESSION_TYPE=filesystem`
 
 ### CLIENT_PUBLIC_KEYS_DIRECTORY
+
 - **Description:** Path to the directory where client public keys are stored. These keys are used for JWT validation.
 - **Example:** `CLIENT_PUBLIC_KEYS_DIRECTORY=keys/public_keys`
 
 ### SERVER_PRIVATE_KEY
+
 - **Description:** Path to the server's private key file. This key is used for creating and signing tokens.
 - **Example:** `SERVER_PRIVATE_KEY=keys/server_private.pem`
 
 ### SERVER_PUBLIC_KEY
+
 - **Description:** Path to the server's public key file. This key is used for verifying tokens.
 - **Example:** `SERVER_PUBLIC_KEY=keys/server_public.pem`
 
 ### SESSION_LIFETIME_MINUTES
+
 - **Description:** Duration (in minutes) after which the session will expire.
 - **Example:** `SESSION_LIFETIME_MINUTES=30`
 
 ### REDIRECT_URL_TO_LOUIS_FRONTEND
+
 - **Description:** The URL to which the Louis login backend will redirect users, leading them to the Louis login frontend where they can provide an email address.
 - **Example:** `REDIRECT_URL_TO_LOUIS_FRONTEND=https://login.louisfrontend.com/`
 
 Once you have defined all these variables, save and close the `.env` file. The Flask application will now use these environment variable values when it runs.
 
 ### 7. Run the Main Flask Application:
+
 With your virtual environment activated, start the main `app.py`:
+
 ```bash
 flask run
 ```
 
 ### 8. Simulate a Client Application:
+
 Open a separate terminal or command prompt. Make sure the virtual environment is activated and then run the `test1app.py` to simulate a client application:
+
 ```bash
 flask --app .\test1app.py run --port=4000
 ```
 
 ### 9. Interact with Louis Login Frontend:
+
 Ensure that the Louis Login Frontend React application is running, ideally on `localhost`. This application will serve as the frontend interface for users to provide their email addresses to Louis Login Backend.
 
 ---
 
 You can now interact with both the main Flask application and the client simulator to validate the entire authentication flow.
+
+## Running the app from dockerfile
+
+### Generate Server and Client Keys
+
+#### Prerequisites
+
+- OpenSSL installed on your machine (in WSL).
+
+#### Steps
+
+1. Navigate to the project's root directory.
+2. Make the script executable:
+
+   ```bash
+   chmod +x generate_keys.sh
```

Comment:
git is able to keep the executation flag so not sure why this is need

Reply:
done